common/tools/eslint-plugin-azure-sdk/src/rules/ts-doc-internal-private-member.ts

```diff
@@ -0,0 +1,131 @@
+import { Rule } from "eslint";
+import { Node } from "estree";
+import { readFileSync } from "fs";
+import { sync as globSync } from "glob";
+import { relative } from "path";
+import { Modifier, SyntaxKind } from "typescript";
+import { getRuleMetaData } from "../utils";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * Helper method for reporting on a node
+ * @param node the Node being operated on
+ * @param context the ESLint runtime context
+ * @param converter a converter from TSESTree Nodes to TSNodes
+ * @param typeChecker the TypeScript TypeChecker
+ * @throws if the Node passes throught the initial checks and has an internal tag
+ */
+const reportInternal = (
+  node: Node,
+  context: Rule.RuleContext,
+  converter: ParserWeakMapESTreeToTSNode,
+): void => {
+  const tsNode = converter.get(node as TSESTree.Node) as any;
+
+  const modifiers = converter.get(node as TSESTree.Node).modifiers;
+  // if type is internal and has a TSDoc and is a private member
+  if (tsNode.jsDoc !== undefined
+    && modifiers !== undefined && modifiers.some(
+      (modifier: Modifier): boolean => modifier.kind === SyntaxKind.PrivateKeyword)
+  ) {
+    // fetch all tags
+    let TSDocTags: string[] = [];
+    tsNode.jsDoc.forEach((TSDocComment: any): void => {
+      TSDocTags = TSDocTags.concat(
+        TSDocComment.tags !== undefined
+          ? TSDocComment.tags.map((TSDocTag: any): string => TSDocTag.tagName.escapedText)
+          : []
+      );
+    });
+
+    // see if any tags match internal
+    if (TSDocTags.some((TSDocTag: string): boolean => /(internal)/.test(TSDocTag))) {
+      context.report({
+        node: node,
+        message: "private class members should not include an @internal tag"
+      });
+    }
+  }
+};
+
+/**
+ * Determine whether this rule should examine a given file
+ * @param fileName the filename of the file in question
+ * @param exclude the list of files excluded by TypeDoc (other than those in node_modues)
+ * @returns false if not in src or is excluded by TypeDoc
+ */
+const shouldExamineFile = (fileName: string, exclude: string[]): boolean => {
+  if (!/src/.test(fileName)) {
+    return false;
+  }
+  const relativePath = relative("", fileName).replace(/\\/g, "/");
+  return !exclude.includes(relativePath);
+};
+
+let exclude: string[] = [];
+try {
+  const typeDocText = readFileSync("typedoc.json", "utf8");
+  const typeDoc = JSON.parse(typeDocText);
+
+  // if typeDoc.exclude exists, add all files matching the glob patterns to exclude
+  if (typeDoc.exclude !== undefined) {
+    typeDoc.exclude.forEach((excludedGlob: string): void => {
+      exclude = exclude.concat(
+        globSync(excludedGlob).filter(
+          (excludeFile: string): boolean => !/node_modules/.test(excludeFile)
+        )
+      );
+    });
+  }
+} catch (err) {
+  exclude = [];
+}
```

Comment:
I do not think we need to worry about typedoc's exclude. @jeremymeng what do you think?

Reply:
I am not sure. It's from ts-doc-internal.ts.  maybe it's an optimization that skips files that we don't care about (node_moduels, src/generated, etc)

Reply:
eslint lints only the folders specified in the executed command. [typedoc's exclude](https://typedoc.org/guides/options/#exclude) seems to tell typedoc to ignore the specified files list during documentation generation, something we should not worry about I believe since all our source files are used to generate the .d.ts file which in turn will eventually be used by the docs team to generate the documentation.

Reply:
@hamsa-s let's delete this code as well.

Reply:
agree that we should not exclude files here.